lucet-runtime/tests/instruction_counting.rs

```diff
@@ -110,6 +110,93 @@ pub fn check_instruction_count() {
+    let files: Vec<DirEntry> = get_instruction_count_test_files();
+
+    assert!(
+        !files.is_empty(),
+        "there are no test cases in the `instruction_counting` directory"
+    );
+
+    files.iter().for_each(|ent| {
+        let wasm_path = ent.path();
+        let do_instrument = true;
+        let module = wasm_test(&wasm_path, do_instrument).expect("can load instrumented module");
+
+        let region = MmapRegion::create(1, &Limits::default()).expect("region can be created");
+
+        let mut inst = region
+            .new_instance(module)
+            .expect("instance can be created");
+
+        // This is quite ugly; it should be better with a proper async interface wrapped around the
+        // low-level `run_bounded` / `resume_bounded` calls.
+        let mut yields = 0;
+        match inst
+            .run_bounded("test_function", &[], 1000)
+            .expect("instance runs")
+        {
+            RunResult::Returned(_) => {
+                // Function returned.
+            }
+            RunResult::Yielded(_) => {
+                panic!("yield unexpected!");
+            }
+            RunResult::BoundExpired => {
+                yields += 1;
+                loop {
+                    match inst.resume_bounded(1000).expect("instance runs") {
+                        RunResult::Returned(_) => {
+                            break;
+                        }
+                        RunResult::Yielded(_) => panic!("yield unexpected!"),
+                        RunResult::BoundExpired => {
+                            yields += 1;
+                            if yields > 1000 {
+                                panic!("Instruction-counting test ran for too long");
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        let instruction_count = inst
+            .get_instruction_count()
+            .expect("instruction count expected from instance");
+
+        assert_eq!(
+            instruction_count,
+            match inst
+                .run("instruction_count", &[])
+                .expect("instance still runs")
+            {
+                RunResult::Returned(value) => value.as_i64() as u64,
+                RunResult::Yielded(_) | RunResult::BoundExpired => {
+                    panic!("instruction counting test runner doesn't support yielding");
+                }
+            },
+            "instruction count for test case {} is incorrect",
+            wasm_path.display()
+        );
+
+        // There is some variance to this in the real world, because yields happen only at the ends
+        // of basic blocks, and there may be an off-by-one; so allow a small range.
+        let expected_yields = instruction_count / 1000;
+        let diff = if expected_yields > yields {
+            expected_yields - yields
+        } else {
+            yields - expected_yields
+        };
```

Comment:
```suggestion
        let diff = (yields - expected_yields).abs();
```
? i think this ought to work, anyway.

Reply:
Unsigned types, so alas I don't think this'll work... it seems there's even an open Rust issue to add something to the stdlib for this: rust-lang/rust#62111.